src/libappimage/utils/ElfFile.cpp

```diff
@@ -0,0 +1,151 @@
+// system
+extern "C" {
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/mman.h>
+#include <sys/types.h>
+}
+
+// local
+#include "light_byteswap.h"
+#include "ElfFile.h"
+
+
+typedef Elf32_Nhdr Elf_Nhdr;
+
+
+#if __BYTE_ORDER == __LITTLE_ENDIAN
+#define ELFDATANATIVE ELFDATA2LSB
+#elif __BYTE_ORDER == __BIG_ENDIAN
+#define ELFDATANATIVE ELFDATA2MSB
+#else
+#error "Unknown machine endian"
```

Comment:
It's "endianness", or, more straightforward, "byte order". Also, this is compiler defined, not machine defined. Just spare "machine" in there, I'd say.

Reply:
Actually, it's the opposite. Processors (machines) use a given endianness and the compiler assemble the applications for a given target machine. By the way, those comments aren't mine. I just ported to c++ the old elf management code.

Reply:
The point is, you can also cross-compile. The endianness is just defined by the target architecture, but not the one on which you compile. My point of view is twofold: during compilation the endinanness is compiler defined, during runtime it's defined by the processor (or emulator, but that's then a virtual processor).

We can improve comments where we see them just fine. I didn't want to imply that's a comment made by you, I can't see what came where in a simple side-by-side diff. Please don't feel offended, that was not my intention.